src/utils/http.tsx

```diff
@@ -0,0 +1,21 @@
+
+export async function get(url: string): Promise<Manifest | Collection | null> {
+  let response: Manifest | Collection | null = null
```

Comment:
This is not generic I'm afraid. Mentioning specific interfaces and using specific error messages should be avoided.

Reply:
Yes, in my latest commit I try to create a generic function to fetch api data. On that commit I try to parse the data based on the content type which I get from the headers of the response received. Regarding the content types of the response I had a look on this [article](https://medium.com/@anatoliiyatsenko/understanding-fetch-api-response-methods-and-the-content-type-header-6dcbe7b24ded)